dashboard/modules/reporter/reporter_agent.py

```diff
@@ -787,6 +907,63 @@ def _record_stats(self, stats, cluster_stats):
                         tags={"ip": ip},
                     )
                 )
+            if total_workers_pss > 0.0:
+                records_reported.append(
+                    Record(
+                        gauge=METRICS_GAUGES["workers_mem_pss"],
+                        value=total_workers_pss,
+                        tags={"ip": ip},
+                    )
+                )
+
+        # Record agent CPU/memory.
+        agent_stats = stats["agent"]
+        if agent_stats:
+            agent_pid = str(agent_stats["pid"])
+            # -- raylet CPU --
+            agent_cpu_usage = float(agent_stats["cpu_percent"]) * 100
+            records_reported.append(
+                Record(
+                    gauge=METRICS_GAUGES["agent_cpu"],
+                    value=agent_cpu_usage,
+                    tags={"ip": ip, "pid": agent_pid},
+                )
+            )
+
+            # -- raylet mem --
+            agent_rss = float(agent_stats["memory_info"].rss) / 1.0e6
+            records_reported.append(
+                Record(
+                    gauge=METRICS_GAUGES["agent_mem"],
+                    value=agent_rss,
+                    tags={"ip": ip, "pid": agent_pid},
+                )
+            )
+            agent_mem_full_info = agent_stats.get("memory_full_info")
+            if agent_mem_full_info is not None:
+                agent_uss = float(agent_mem_full_info.uss) / 1.0e6
```

Comment:
nit: feel like some of this common code across different components could be refactored.

Reply:
done